safesocket/pipe_windows.go

```diff
@@ -3,16 +3,27 @@
 
 package safesocket
 
+//go:generate go run golang.org/x/sys/windows/mkwinsyscall -output zsyscall_windows.go pipe_windows.go
+
 import (
+	"context"
 	"fmt"
 	"net"
+	"runtime"
 	"syscall"
+	"time"
 
-	"github.com/Microsoft/go-winio"
+	"github.com/tailscale/go-winio"
+	"golang.org/x/sys/windows"
 )
 
 func connect(s *ConnectionStrategy) (net.Conn, error) {
-	return winio.DialPipe(s.path, nil)
+	dl := time.Now().Add(2 * time.Second)
```

Comment:
this enough? Our GitHub Actions CI has shown me that even huge timeouts like this regularly trip 😭

Reply:
It's what the previous `go-winio` API did under the hood. Happy to make this longer!

Reply:
Naah, this is fine if it's not a change and hasn't caused us CI flakes this far.

Reply:
I bumped to 20 for now.